lib/fastifySession.js

```diff
@@ -106,54 +110,49 @@ function session (fastify, options, next) {
         done()
         return
       }
-      let sessionId = request.cookies[cookieName]
-      if (sessionId && hasCookiePrefix && sessionId.startsWith(cookiePrefix)) {
-        sessionId = sessionId.slice(cookiePrefixLength)
-      }
-      if (!sessionId) {
+
+      const cookieSessionId = getCookieSessionId(request)
+      if (!cookieSessionId) {
         request.session = new Session(request, idGenerator, cookieOpts, cookieSigner)
         done()
       } else {
-        decryptSession(sessionId, options, request, done)
+        decryptSession(cookieSessionId, options, request, done)
       }
     }
   }
 
   function onSend (options) {
     const cookieOpts = options.cookie
-    const cookieName = options.cookieName
-    const cookiePrefix = options.cookiePrefix
```

Comment:
Why is all the cookie prefix code gone?

Reply:
I cleaned it up a bit. See this commit to see the isolated changes:
https://github.com/fastify/session/pull/149/commits/7eba911de77054fb0cf3503b7c18776cf50d51e1

After cleaning, I noticed that this line was a potential problem:
https://github.com/fastify/session/blob/master/lib/fastifySession.js#L143

`request.cookies[cookieName]` would still have the prefix so it shouldn't be compared to `encryptedSessionId` directly.